cmd/microcloud/session.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/canonical/microcloud/microcloud/api/types"
+	cloudClient "github.com/canonical/microcloud/microcloud/client"
+	"github.com/canonical/microcloud/microcloud/mdns"
+	"github.com/canonical/microcloud/microcloud/service"
+)
+
+type SessionRole string
+
+const (
+	SessionInitiating SessionRole = "initiating"
+	SessionJoining    SessionRole = "joining"
+)
+
+type SessionFunc func(gw *cloudClient.WebsocketGateway) error
+
+func (c *initConfig) runSession(ctx context.Context, s *service.Handler, role SessionRole, timeout time.Duration, f SessionFunc) error {
+	cloud := s.Services[types.MicroCloud].(*service.CloudService)
+	conn, err := cloud.StartSession(ctx, string(role), timeout)
+	if err != nil {
+		return err
+	}
+
+	defer conn.Close()
+
+	return f(cloudClient.NewWebsocketGateway(ctx, conn))
+}
+
+func (c *initConfig) initiatingSession(gw *cloudClient.WebsocketGateway, sh *service.Handler, services []types.ServiceType, passphrase string, expectedSystems []string) error {
+	session := types.Session{
+		Address:    c.address,
+		Interface:  c.lookupIface.Name,
+		Services:   services,
+		Passphrase: passphrase,
+	}
+
+	err := gw.Write(session)
+	if err != nil {
+		return fmt.Errorf("Failed to send session start: %w", err)
+	}
+
+	err = gw.ReceiveCombined(&session)
+	if err != nil {
+		return fmt.Errorf("Failed to read session reply: %w", err)
+	}
+
+	if !c.autoSetup {
+		fmt.Printf("Enter the passphrase on systems that you want to join the cluster:\n\n %s\n\n", session.Passphrase)
+		fmt.Println("Waiting for systems to send their intent to join ...")
```

Comment:
I have changed the dialog to this:

```
Select an address for MicroCloud's internal traffic:

 Using address "10.165.133.71" for MicroCloud

Use the following command on systems that you want to join the cluster:

 microcloud join

When requested enter the passphrase:

 juice imaging exterior nevermore

Waiting for systems to send their intent to join ...
```

Reply:
Fixed

Reply:
But I am still wondering if we should maybe put the used fingerprint in another line.

Reply:
Yeah I think maybe this can be
```
Use the following command on systems that you want to join the cluster:
 
  microcloud join

When requested enter the passphrase:

  urgent crumble emperor governor

Verify the fingerprint is displayed on joining systems: f211051397df
Waiting to detect systems (because they won't be joining quite yet :)) ...
```

Reply:
Mh yeah this looks better!

I wen with `Verify the fingerprint "30a09aeae52e" is displayed on joining systems.` as the remainder of the dialog also tends to provide the quoted info inline and not after a colon.